lib/idempo.rb

```diff
@@ -58,6 +59,12 @@ def call(env)
 
       expires_in_seconds = (headers.delete("X-Idempo-Persist-For-Seconds") || @persist_for_seconds).to_i
       if response_may_be_persisted?(status, headers, body)
+        if Gem::Version.new(Rack.release) >= Gem::Version.new("3.0")
+          # `body` could be of type `ActionDispatch::Response::RackBody` and idempo will not even attempt to store it,
```

Comment:
Let's alter this comment because it is a bit misleading. A Rack response body _may permit_ "materialization" into an Array of strings using `to_ary` - this is in fact in the Rack spec, here https://github.com/rack/rack/blob/main/SPEC.rdoc#the-body- So in this case, checking for `respond_to?(:to_ary)` and calling it is sufficient. For `each` and `call` we might need more special cases, but in this instance it is enough to state that some Rack bodies can be "expanded" using `to_ary`. It doesn't matter whether such a body comes from Rails or any other callee. It is also important to note that if we did use `to_ary` we are not supposed to call `close` - i.e. we need to stick to the Rack semantics very carefully here, and the comments should reflect that.